### ISSUE TITLE
merge list 

### ISSUE THREAD

**Issue by wolgod** [peterbourgon/mergemap#6]
in my case 
{
			src:      `{"1": { "b":1, "2": { "3": {         "b":3, "n":[{"aaa":"aaa"},{"bbb":"bbb"}]} }        }}`,
			dst:      `{"1": {        "2": { "3": {"a":"A",        "n":":[{"ccc":"ccc"}]"} }, "a":3 }}`,
			expected: `{"1": { "b":1, "2": { "3": {"a":"A", "b":3, "n":[{"aaa":"aaa"},{"bbb":"bbb"},{"ccc":"ccc"}]} }, "a":3 }}`,
},
 I think it expectd result will true 

but the result is {"1": { "b":1, "2": { "3": {"a":"A", "b":3, "n":[{"aaa":"aaa"},{"bbb":"bbb"]} }, "a":3 }}

**Comment by peterbourgon** [comment]
```
"n":":[{"ccc":"ccc"}]"
```

Is this correct?

**Comment by peterbourgon** [comment]
Simplifying the example, let's assume `src` is always:

```json
{"slice": [{"a": "a"}, {"b": "b"}]}
```

and that we merge with a given `dst` to get expected `res`.

So your example sets `dst` as:

```json
{"slice": [{"c": "c"}]}
```

in which case the result of the merge is probably this:

```json
{"slice": [{"a": "a"}, {"b": "b"}, {"c": "c"}]}
```

But what about if we set this as dst:

```json
{"slice": [{"a": "X"}]}
```

Which one of these is the correct result of the merge?

```json
{"slice": [{"a": "a"}, {"b": "b"}, {"a": "X"}]}
{"slice": [{"a": "X"}, {"b": "b"}            ]}
{"slice": [            {"b": "b"}, {"a": "X"}]}
```

Or what about merging with:

```json
{"slice": [{"b": "b"}]}
```

Which one of these is the correct result?

```json
{"slice": [{"a": "a"}, {"b": "b"}            ]}
{"slice": [{"a": "a"}, {"b": "b"}, {"b": "b"}]}
```

Or what about merging with:

```json
{"slice": [{"b": "c", "x": "y"}]}
```

Which one of these is the correct result?

```json
{"slice": [{"a": "a"}, {"b": "b"}                                ]}
{"slice": [{"a": "a"},                       {"b": "c", "x": "y"}]}
{"slice": [{"a": "a"}, {"b": "b"},           {"b": "c", "x": "y"}]}
{"slice": [{"a": "a"}, {"b": "c", "x": "y"}                      ]}
```

tl;dr: I don't think slices have obvious merge semantics

**Comment by peterbourgon** [comment]
Closing for now. Happy to hear different thoughts, though!
